java_src/foam/nanos/pm/DAOPMLogger.java

```diff
@@ -16,10 +16,11 @@
   implements PMLogger, NanoService
 {
 
+  MapDAO dao_;
+
   @Override
   public void log(PM pm) {
-    MapDAO dao = (MapDAO) getX().get("pminfodao");
-    PMInfo pmi = (PMInfo) dao.find(pm);
+    PMInfo pmi = (PMInfo) dao_.find(pm);
```

Comment:
This line won't work. 'pm' isn't PIMinfo's key. You would need something like:
find(AND(EQ(PMInfo.CLSNAME, pm.classname), EQ(PMInfo.PMNAME, pm.pmname)))